__tests__/integration/multisig.test.js

```diff
@@ -0,0 +1,276 @@
+    expect(sig6).toBeTruthy();
+
+    // try to send
+    response = await TestUtils.request
+      .post('/wallet/tx-proposal/sign-and-push')
+      .send({txHex, signatures: [sig1, sig2, sig3, sig4, sig5, sig6]})
+      .set({ 'x-wallet-id': wallet1.walletId });
+    console.log(JSON.stringify(response.body));
+
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(false);
+    done();
+  });
+
+  it('Should send a transaction with minimum signatures', async done => {
+    const tx = {
+      input: [fundTx1],
+      outputs: [
+        { address: "WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc", value: 100 },
+        { address: "wcUZ6J7t2B1s8bqRYiyuZAftcdCGRSiiau", value: 270 },
+      ],
+    };
+
+    // wallet1 proposes the transaction
+    let response = await TestUtils.request
+      .post('/wallet/tx-proposal')
+      .send(tx)
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    const txHex = response.body.txHex;
+
+    // collect signatures from 3 wallets
+    const sig1 = await wallet1.getSignatures(txHex);
+    expect(sig1).toBeTruthy();
+    const sig2 = await wallet2.getSignatures(txHex);
+    expect(sig2).toBeTruthy();
+    const sig3 = await wallet3.getSignatures(txHex);
+    expect(sig3).toBeTruthy();
+
+    // try to send
+    response = await TestUtils.request
+      .post('/wallet/tx-proposal/sign-and-push')
+      .send({txHex, signatures: [sig1, sig2, sig3]})
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.hash).toBeDefined();
+    done();
+  });
+
+  it('Should send a transaction with max signatures', async done => {
+    const tx = {
+      outputs: [
+        { address: "WPynsVhyU6nP7RSZAkqfijEutC88KgAyFc", value: 1 },
+        { address: "wcUZ6J7t2B1s8bqRYiyuZAftcdCGRSiiau", value: 2 },
+      ],
+    };
+
+    // wallet1 proposes the transaction
+    let response = await TestUtils.request
+      .post('/wallet/tx-proposal')
+      .send(tx)
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    const txHex = response.body.txHex;
+
+    // collect signatures from 5 wallets
+    const sig1 = await wallet1.getSignatures(txHex);
+    expect(sig1).toBeTruthy();
+    const sig2 = await wallet2.getSignatures(txHex);
+    expect(sig2).toBeTruthy();
+    const sig3 = await wallet3.getSignatures(txHex);
+    expect(sig3).toBeTruthy();
+    const sig4 = await wallet4.getSignatures(txHex);
+    expect(sig4).toBeTruthy();
+    const sig5 = await wallet5.getSignatures(txHex);
+    expect(sig5).toBeTruthy();
+
+    // try to send
+    response = await TestUtils.request
+      .post('/wallet/tx-proposal/sign-and-push')
+      .send({txHex, signatures: [sig1, sig2, sig3, sig4, sig5]})
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.hash).toBeDefined();
+    done();
+  });
+});
```

Comment:
When we run `get-my-signatures` with the `walletExtra` it will return just the pubkey since it does not recognize any inputs as being from the wallet. So when creating the input data it will actually test the less than min signatures scenario.

Reply:
Done in [79f1802](https://github.com/HathorNetwork/hathor-wallet-headless/pull/156/commits/79f1802633d0261c0dab63058f047ddf009bfd06)